src/Build/Graph/ProjectGraph.cs

```diff
@@ -75,6 +76,16 @@ public sealed class ProjectGraph
 
         public IReadOnlyCollection<ProjectGraphNode> GraphRoots { get; }
 
+        /// <summary>
+        ///     Gets a distinct enumeration of all of the environment variables reads during evaluation.
+        /// </summary>
+        public IEnumerable<string> EnvironmentVariableReads => _projectNodesTopologicallySorted?.Value.Select(p => p.ProjectInstance).SelectMany(pi => pi.EnvironmentVariableReads).Distinct();
```

Comment:
>_projectNodesTopologicallySorted [](start = 63, length = 32)

Under what circumstances this is null?
Can we return an empty collection if that's the case instead of null? I'd prefer to avoid forcing consumers to do null checks all over the place

Reply:
I added them as a simple check, but am not aware of the current (or desired) behavior.